torch/distributions/transforms.py

```diff
@@ -90,30 +88,24 @@ def __call__(self, x):
         """
         Computes the transform `x => y`.
         """
-        return self._call(x)
-
-    def _inv_call(self, y):
-        """
-        Inverts the transform `y => x`.
-        """
-        return self._inverse(y)
-
-    def _cached_call(self, x):
-        """
-        Computes the memoized transform `x => y`.
-        """
-        x_old, y_old = self._cached_x_y
+        try:
+            x_old, y_old = self._cached_x_y
+        except AttributeError:
+            return self._call(x)
```

Comment:
This will be very slow if you don't use caches. Maybe just add a flag and use an `if`?